### PR TITLE
[Attention] add gpt-oss support

### PR DESCRIPTION
**Pull request by taoyao1221**
### What this PR does / why we need it?

This PR adapts the gpt-oss model and implements the SWA and Sink features required by gpt-oss based on the torch_npu.npu_fused_infer_attention_score_v2 fusion operator.

### Does this PR introduce _any_ user-facing change?
1. Modified the code in the Attention section to adapt to the SWA and Sink features required by gpt-oss.
2. Modified the code in the MoE section to add support for bias and swigluoai.

### How was this patch tested?

image
The eager mode and Piecewise ACLGraph mode were tested and verified using evalscope.
Through optimization with the torch_npu.npu_fused_infer_attention_score_v2 fusion operator and graph mode, simple performance tests were conducted. Compared to the small operator mode, under 10 concurrent requests, input size of 128, and output size of 128:
TTFT: 810ms (small operator mode) --> 228ms (FIA eager mode) --> 217ms (Piecewise ACLGraph mode);
TPOT: 512ms (small operator mode) --> 79ms (FIA eager mode) --> 20ms (Piecewise ACLGraph mode).

- vLLM version: v0.12.0
- vLLM main: https://github.com/vllm-project/vllm/commit/ad32e3e19ccf0526cb6744a5fed09a138a5fb2f9

### CODE REVIEW

**Comment by wangxiyuan** [comment]
please fix the merge conflict and add e2e test for gpt-oss

**Comment by weijinqian0** [comment]
need rebase

File: vllm_ascend/attention/attention_v1.py [hunk 1]
```diff
@@ -563,6 +564,19 @@ def __init__(
...
         self.value_cache = None
+        self.share_mask_tril_spase = None
+        if self.sliding_window is not None:
+            mask = torch.ones(2048, 2048, dtype=torch.bool)
+            triu_mask = torch.triu(mask, diagonal=1).to("npu")
+            tril_mask = torch.tril(mask, -self.sliding_window).to("npu")
+            self.share_mask_tril_spase = triu_mask + tril_mask
+        else:
+            self.share_mask_tril_spase = ~torch.tril(
+                torch.ones((2048, 2048), device='npu', dtype=torch.bool))
```
Comment (MengqingCao) [inline vllm_ascend/attention/attention_v1.py]: Let's expand `AttentionMaskBuilder` to support sliding window instead of creating mask here
https://github.com/vllm-project/vllm-ascend/blob/main/vllm_ascend/attention/attention_mask.py#L32

File: vllm_ascend/ops/fused_moe/moe_mlp.py [hunk 2]
```diff
@@ -234,14 +237,31 @@ def unquant_apply_mlp(hidden_states: torch.Tensor,
...
+    # Do activation
+    def swiglu_oai(gate_up):
+        alpha = 1.702
+        limit = 7.0
+        gate, up = gate_up[..., ::2], gate_up[..., 1::2]
+        gate = gate.clamp(min=None, max=limit)
+        up = up.clamp(min=-limit, max=limit)
+        glu = gate * torch.sigmoid(gate * alpha)
+        gated_output = (up + 1) * glu
+        return gated_output
```
Comment (weijinqian0) [inline vllm_ascend/ops/fused_moe/moe_mlp.py]: Move this function to activation.py.

File: vllm_ascend/ops/fused_moe/moe_mlp.py [hunk 3]
```diff
@@ -234,14 +237,31 @@ def unquant_apply_mlp(hidden_states: torch.Tensor,
...
         weight=[w1],
+        bias=[w1_bias.to(
+            dtype=torch.float32)] if w1_bias is not None else None,
         split_item=2,
         group_list_type=group_list_type,
         group_type=0,
         group_list=group_list,
     )[0]
+
+    # Do activation
```
Comment (weijinqian0) [inline vllm_ascend/ops/fused_moe/moe_mlp.py]: may need transfer to triton

**Comment by wangxiyuan** [comment]
Any progress? If this PR is still alive, please rebase to main and make CI happy, otherwise you can close it. Thanks

And please add a e2e test to ensure it works.
